projects/packages/forms/src/blocks/contact-form/edit.js

```diff
@@ -112,9 +128,25 @@ function JetpackContactFormEdit( { name, attributes, setAttributes, clientId, cl
 		customThankyouRedirect,
 		formTitle,
 		variationName,
+		name: formName,
+		uuid,
 	} = attributes;
 	const instanceId = useInstanceId( JetpackContactFormEdit );
 
+	// Effect for UUID generation
+	useEffect( () => {
+		if ( ! uuid ) {
+			setAttributes( { uuid: generateUUID() } );
+		}
+	}, [ uuid, setAttributes ] );
+
+	// Effect for name generation
+	useEffect( () => {
+		if ( ! formName ) {
+			setAttributes( { name: generateRandomName() } );
+		}
+	}, [ formName, setAttributes ] );
```

Comment:
If the name is random, and also not unique, what's the reason we need it? I know I'm missing some convo or context.

Reply:
Those are all valid concerns, we've had some light discussion ( ref: p1753133273097319/1753121466.468139-slack-C052XEUUBL4 ) on why these should exist and it's true it doesn't make full sense right now. The name might also not be random, but inherit from the post title (which is what we usually do), but the intention of the uuid/name pair is that it represents a form with its identifier and a name for the user to change. This implementation is quite rough as we won't be offering any UI for the user to change this at the moment.